core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -809,9 +825,14 @@ private[spark] class TaskSetManager(
         }
       }
     }
-    // Also re-enqueue any tasks that were running on the node
     for ((tid, info) <- taskInfos if info.running && info.executorId == execId) {
-      handleFailedTask(tid, TaskState.FAILED, ExecutorLostFailure(execId))
+      // Also re-enqueue any tasks that were running on the node
+      val executorFailureReason = reason match {
+        case exited: ExecutorExitedNormally =>
```

Comment:
This would go away if you follow my suggestion of merging the two errors, but in any case, since it's a case class:

    case ExecutorExitedNormally(exitCode, reason) =>